app/controllers/push_notifications_controller.rb

```diff
@@ -0,0 +1,55 @@
+require_dependency 'push_notifications'
+
+class PushNotificationsController < ApplicationController
+  layout false
+  before_filter :ensure_logged_in
+  skip_before_filter :preload_json
+
+  def subscribe
+    endpoint = params[:endpoint]
+
+    if endpoint.start_with?(PushNotifications::GCM_ENDPOINT)
+      PushNotifications.subscribe(current_user, endpoint)
+      render json: success_json
+    else
+      render json: failed_json
+    end
+  end
+
+  def unsubscribe
+    PushNotifications.unsubscribe(current_user)
+    render json: success_json
+  end
+
+  def latest
```

Comment:
Payload hasn't been released for push messaging so we have to do this. See https://code.google.com/p/chromium/issues/detail?id=486040